### ISSUE TITLE
Upgrading from LoadBalancer to NodePort service for the meshgateways leaves federation secret with old load balancer address

### ISSUE THREAD

**Issue by ishustava** [hashicorp/consul-k8s#417]
### Overview of the Issue

Upgrading from the LoadBalancer to NodePort service type for the mesh gateways, leaves federation secret with both the old load balancer address and the new NodePort address.

#### Install the helm chart with the following config
```
global:
  name: consul
  tls:
    enabled: true
  federation:
    enabled: true
    createFederationSecret: true
connectInject:
  enabled: true
meshGateway:
  enabled: true
```

```
helm install consul -f config.yaml hashicorp/consul
```

Wait for everything to come up and check that the federation secret has mesh gateway's load balancer address.

#### Upgrade with the following config:
```
global:
  name: consul
  tls:
    enabled: true
  federation:
    enabled: true
    createFederationSecret: true
connectInject:
  enabled: true
meshGateway:
  enabled: true
  service:
    type: NodePort
    nodePort: 30000
```

```
helm ugrade consul -f config.yaml hashicorp/consul
```

Check the federation secret once the upgrade is finished

```
$ kubectl get secret consul-federation -o jsonpath='{.data.serverConfigJSON}' | base64 -d
{"primary_datacenter":"dc1","primary_gateways":["40.88.229.151:443","172.16.1.6:30000"]}
```

### Expected behavior

The federation secret should contain only the node port address since the load balancer is no longer reachable.

### Environment details

If not already included, please provide the following:
- `consul-k8s` version: the one included with consul-helm
- `consul-helm` version: 0.28.0

Any other information you can provide about the environment/deployment.

### Additional Context

This problem can be worked around by running a helm upgrade command again, which will cause the federation secret to be updated with the right addresses.

**Comment by ishustava** [comment]
Duplicate of #366
